pkg/utils/registry/registry.go

```diff
@@ -18,12 +18,33 @@ package registry
 
 import (
 	"fmt"
+	"sync"
 
 	"github.com/labring/sealos/fork/github.com/heroku/docker-registry-client/registry"
 	"github.com/labring/sealos/pkg/utils/logger"
 )
 
+type option struct {
+	url, username, password string
+	skipTLS                 bool
+}
+
+var (
+	sharedClients = make(map[*option]*registry.Registry)
+	lock          sync.RWMutex
+)
+
 func NewRegistry(url, username, password string, skipTLS bool) (*registry.Registry, error) {
+	key := &option{url, username, password, skipTLS}
+	lock.Lock()
+	defer lock.Unlock()
+	if v, ok := sharedClients[key]; ok {
```

Comment:
This doesn't always seem ok, you should replace the pointer type with a struct type, which will compare each property value of the struct.

Reply:
> This doesn't always seem ok, you should replace the pointer type with a struct type, which will compare each property value of the struct.

make sense👍